src/bitwise.rs

```diff
@@ -57,13 +59,43 @@ impl Bits for u32 {
     fn get_bit(self, bit_idx: u8) -> bool {
         self.is_bit_on(bit_idx)
     }
+
+    /// To clear all bits from the `most significant bit` through `bit_idx` (inclusive)
+    fn clear_bits_msb_to_idx(&mut self, bit_idx: u8) {
+        debug_assert!(bit_idx < 32);
+
+        let mask = (1 << bit_idx) - 1;
+        *self &= mask;
+    }
+
+    /// To clear all bits from `bit_idx` through `0` (inclusive)
+    fn clear_bits_idx_to_0(&mut self, bit_idx: u8) {
+        debug_assert!(bit_idx < 32);
+
+        let mask = -1_i32 << (bit_idx + 1);
+        *self &= mask as u32;
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use rand::Rng;
 
+    #[test]
+    fn test_clear_bits_msb_to_idx() {
+        let mut b = 0b11111111;
+        b.clear_bits_msb_to_idx(3);
+        println!("{}", b)
+    }
+
+    #[test]
+    fn test_clear_bits_idx_to_0() {
+        let mut b = 0b11111;
+        b.clear_bits_idx_to_0(3);
+        println!("{}", b)
+    }
```

Comment:
Tests without assertion? 🤔

Reply:
I'm sorry, I had a big problem afternoon and i didn't check well.